metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml

```diff
@@ -65,6 +65,102 @@
         </value-attributes>
     </property>
     <property>
+        <name>metron_log_dir</name>
+        <value>/var/log/metron</value>
+        <description>Log directory for metron</description>
+        <display-name>Metron log dir</display-name>
+    </property>
+    <property>
+        <name>metron_pid_dir</name>
+        <value>/var/run/metron</value>
+        <description>The directory for metron pid files</description>
+        <display-name>Metron pid dir</display-name>
+    </property>
+    <property>
+        <name>metron_rest_port</name>
+        <value>8082</value>
+        <description>Port the REST application runs on</description>
+        <display-name>Metron REST port</display-name>
+    </property>
+    <property>
+        <name>metron_jvm_flags</name>
+        <description>JVM flags passed to Metron scripts</description>
+        <display-name>Metron JVM flags</display-name>
+        <value/>
+        <value-attributes>
+            <empty-value-valid>true</empty-value-valid>
+        </value-attributes>
+    </property>
+    <property>
+        <name>metron_spring_profiles_active</name>
+        <description>Active Spring profiles</description>
+        <display-name>Active Spring profiles</display-name>
+        <value/>
+        <value-attributes>
+            <empty-value-valid>true</empty-value-valid>
+        </value-attributes>
+    </property>
+    <property require-input="true">
+        <name>metron_jdbc_driver</name>
+        <value></value>
+        <description>Class name of the JDBC Driver used by Metron</description>
+        <display-name>Metron JDBC Driver</display-name>
+    </property>
+    <property require-input="true">
+        <name>metron_jdbc_url</name>
+        <value></value>
+        <description>JDBC Connection URL used by Metron</description>
+        <display-name>Metron JDBC URL</display-name>
+    </property>
+    <property require-input="true">
+        <name>metron_jdbc_username</name>
+        <value></value>
+        <description>Metron JDBC Username</description>
+        <display-name>Metron JDBC username</display-name>
+    </property>
+    <property require-input="true">
```

Comment:
I'm not particularly familiar with it, but Ambari has some setup for handling passwords.  We should look into what their recommended way of handling them is, so it's setup right.

Super brief digging has led to a couple things in hive.
```
<property require-input="true">
    <name>javax.jdo.option.ConnectionPassword</name>
    <value> </value>
    <property-type>PASSWORD</property-type>
    <display-name>Database Password</display-name>
    <description>password to use against metastore database</description>
    <value-attributes>
      <type>password</type>
      <overridable>false</overridable>
      <hidden>HIVE_CLIENT,WEBHCAT_SERVER,HCAT,CONFIG_DOWNLOAD</hidden>
    </value-attributes>
    <on-ambari-upgrade add="true"/>
```
eventually this becomes 
```
      create_schema_cmd = format("export HIVE_CONF_DIR={hive_server_conf_dir} ; "
                                 "{hive_schematool_bin}/schematool -initSchema "
                                 "-dbType {hive_metastore_db_type} "
                                 "-userName {hive_metastore_user_name} "
                                 "-passWord {hive_metastore_user_passwd!p} -verbose")
```
Notably with the `!p` suffix on the password, which I'm sure does some magic.